src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,446 @@
+                "\tDays before password expires: %d -> %d",
+                $fieldSettings['PasswordTTL'],
+                $passwordTTL
+            ));
+        }
+
+        //  enforce CT to use password expiration feature
+        $validatorConfiguration['PasswordValueValidator']['requireNewPassword'] =
+            self::REQUIRE_NEW_PASSWORD_VALUE;
+        $fieldSettings['PasswordTTL'] = $passwordTTL;
+
+        $this->doUpdateContentType(
+            $contentType,
+            $fieldSettings,
+            $validatorConfiguration,
+            $userFieldDefinition
+        );
+    }
+
+    /**
+     * @param array<string, mixed> $fieldSettings
+     * @param array<string, mixed> $validatorConfiguration
+     */
+    private function doUpdateContentType(
+        ContentType $contentType,
+        array $fieldSettings,
+        array $validatorConfiguration,
+        FieldDefinition $fieldDefinition
+    ): void {
+        $this->repository->sudo(
+            function () use (
+                $contentType,
+                $fieldSettings,
+                $validatorConfiguration,
+                $fieldDefinition
+            ): void {
+                $contentTypeDraft = $this->contentTypeService->createContentTypeDraft($contentType);
+                $fieldDefinitionUpdateStruct = $this->contentTypeService->newFieldDefinitionUpdateStruct();
+
+                $fieldDefinitionUpdateStruct->fieldSettings = $fieldSettings;
+                $fieldDefinitionUpdateStruct->validatorConfiguration = $validatorConfiguration;
+
+                $this->contentTypeService->updateFieldDefinition(
+                    $contentTypeDraft,
+                    $fieldDefinition,
+                    $fieldDefinitionUpdateStruct
+                );
+
+                $this->contentTypeService->publishContentTypeDraft($contentTypeDraft);
+            }
+        );
+    }
+
+    /**
+     * @param array<string> $processedContentTypes
+     */
+    private function doesContentTypeNeedUpdate(ContentType $contentType, array $processedContentTypes): bool
+    {
+        $fields = $contentType->getFieldDefinitionsOfType(self::USER_FIELDTYPE_IDENTIFIER);
+        $count = $fields->count();
+
+        if ($count !== 1) {
+            throw new InvalidArgumentException(sprintf(
+                'Expected exactly 1 "%s" field type in "%s" Content Type, found %d',
+                self::USER_FIELDTYPE_IDENTIFIER,
+                $contentType->identifier,
+                $count
+            ));
+        }
+
+        $fieldDefinition = $fields->first();
+
+        $validatorConfiguration = $fieldDefinition->getValidatorConfiguration();
+        $fieldSettings = $fieldDefinition->getFieldSettings();
+
+        $isUpdateNeeded = !$validatorConfiguration['PasswordValueValidator']['requireNewPassword']
+            || 0 === $fieldSettings['PasswordTTL'];
+        $isContentTypeAlreadyProcessed = in_array($contentType->identifier, $processedContentTypes, true);
```

Comment:
This can be established way earlier (at the beginning of the method) and used as an early return.

Reply:
Alright, fixed in [50cf4fa](https://github.com/ezsystems/ezplatform-kernel/pull/347/commits/50cf4fad6a44ff4f50320b7b3551aab0ba2d6670)